oemoflex/model/postprocessing.py

```diff
@@ -0,0 +1,671 @@
+
+    summed_variable_costs = set_index_level(
+        summed_variable_costs,
+        level='var_name',
+        value='summed_variable_costs'
+    )
+
+    return summed_variable_costs
+
+
+def set_index_level(series, level, value):
+    r"""
+    Sets a value to a multiindex level. If the level does not exist, it
+    is appended.
+
+    Parameters
+    ----------
+    series : pd.Series
+
+    level : str
+        Name of the level
+
+    value : str
+        Value to set
+
+    Returns
+    -------
+    series : pd.Series
+        Series with level set to value or appended level with value.
+    """
+    level_names = list(series.index.names)
+
+    series.index.names = level_names
+
+    df = pd.DataFrame(series)
+
+    df.reset_index(inplace=True)
+
+    df[level] = value
+
+    if level not in level_names:
+        level_names.append(level)
+
+    df.set_index(level_names, inplace=True)
+
+    series = df.loc[:, 0]
+
+    return series
+
+
+def filter_by_var_name(series, var_name):
+
+    filtered_ids = series.index.get_level_values(2) == var_name
+
+    filtered_series = series.loc[filtered_ids]
+
+    return filtered_series
+
+
+def map_var_names(scalars):
+
+    def get_component_id(id):
+
+        component_id = get_component_id_in_tuple((id[0], id[1]))
+
+        return component_id
+
+    def get_carrier(id):
+        bus = get_bus_from_oemof_tuple((id[0], id[1]))
+
+        if bus:
+            carrier = str.split(bus.label, '-')[1]
+
+            return carrier
+
+    def get_in_out(id, component_id):
+
+        if not id[1] is np.nan:
+            in_out = ['out', 'in'][component_id]
+
+            return in_out
+
+    def map_index(id):
+        component_id = get_component_id(id)
+
+        component = id[component_id]
+
+        carrier = get_carrier(id)
+
+        in_out = get_in_out(id, component_id)
+
+        var_name = [id[2], in_out, carrier]
+
+        var_name = [item for item in var_name if item is not None]
+
+        var_name = '_'.join(var_name)
+
+        index = (component, None, var_name)
+
+        return index
+
+    scalars.index = scalars.index.map(map_index)
+
+    scalars.index = scalars.index.droplevel(1)
+
+    scalars.index.names = ('name', 'var_name')
+
+    return scalars
+
+
+def add_component_info(scalars):
+
+    scalars.name = 'var_value'
+
+    scalars = pd.DataFrame(scalars)
+
+    scalars['region'] = scalars.index.get_level_values(0).map(lambda x: x.region)
+
+    scalars['type'] = scalars.index.get_level_values(0).map(lambda x: x.type)
+
+    scalars['carrier'] = scalars.index.get_level_values(0).map(lambda x: x.carrier)
+
+    scalars['tech'] = scalars.index.get_level_values(0).map(lambda x: x.tech)
+
+    return scalars
+
+
+def group_by_element(scalars):
+    elements = {}
+    for group, df in scalars.groupby(['carrier', 'tech']):
+        name = '-'.join(group)
+
+        df = df.reset_index()
+
+        df = df.pivot(
```

Comment:
There is a problem with the postprocessing of results when transmission links are involved. Pivot fails because of duplicate entries in index.